datadog/dogstatsd/base.py

```diff
@@ -362,12 +374,25 @@ def open_buffer(self, max_buffer_size=None):
     def close_buffer(self):
         """
         Flush the buffer and switch back to single metric packets.
+
+        Note: This method must be called after a matching open_buffer()
+        invocation.
         """
-        self._send = self._send_to_server
 
-        if self.buffer:
-            # Only send packets if there are packets to send
-            self._flush_buffer()
+        try:
+            if not hasattr(self, 'buffer'):
+                raise BufferError('Cannot close buffer that was never opened')
```

Comment:
That might be correct though it looks like a different change.

Reply:
The edge case was found during the test writing. It's a separate commit but it can be split out into a separate PR if you want.